tests/components/notify/test_demo.py

```diff
@@ -78,12 +80,14 @@ def test_no_notify_service(self, mock_demo_get_service):
     @patch('homeassistant.components.notify.demo.get_service')
     def test_discover_notify(self, mock_demo_get_service):
         """Test discovery of notify demo platform."""
+        assert notify.DOMAIN not in self.hass.config.components
         discovery.load_platform(
-            self.hass, 'notify', 'demo', {'test_key:test_val'}, CONFIG)
+            self.hass, 'notify', 'demo', {'test_key:test_val'}, {})
```

Comment:
Need to fix this: I've made a typo in the discovery info so it's not a dictionary, but a set.